java/bogdana_maksiuta/src/com/zoolatech/lecture3/tasks/_4/Reporter.java

```diff
@@ -0,0 +1,68 @@
+package com.zoolatech.lecture3.tasks._4;
+
+import java.util.Scanner;
+
+public abstract class Reporter {
+    String headerData;
+    String fileData;
+    String fileName;
+
+    public Reporter(String headerData, String fileData, String fileName) {
+        this.headerData = headerData;
+        this.fileData = fileData;
+        this.fileName = fileName;
+    }
+
+    void generate() {
+        System.out.println("\nWhat do you want? \nPress '1' for create a file \nPress '2' for add header data" +
```

Comment:
The method generate doesn't need to accept user input. It needs to perform all tasks (create a new file, add header data in an appropriate format (CSV, PDF, DOC), add file data in an appropriate format and close a file) when it's called.

Reply:
fixed